classes/Dispatcher.php

```diff
@@ -219,25 +219,16 @@ protected function __construct(SymfonyRequest $request = null)
     {
         $this->setRequest($request);
 
-        $this->use_routes = (bool)Configuration::get('PS_REWRITING_SETTINGS');
-
-        // Select right front controller
-        if (defined('_PS_ADMIN_DIR_')) {
-            $this->front_controller = self::FC_ADMIN;
-            $this->controller_not_found = 'adminnotfound';
-        } elseif (Tools::getValue('fc') == 'module') {
-            $this->front_controller = self::FC_MODULE;
-            $this->controller_not_found = 'pagenotfound';
-        } else {
-            $this->front_controller = self::FC_FRONT;
-            $this->controller_not_found = 'pagenotfound';
-        }
+        $this->use_routes = (bool) Configuration::get('PS_REWRITING_SETTINGS');
 
+        $frontControllerType = $this->setFrontControllerType();
```

Comment:
Maybe we can rename it in `defineControllerType` because even if you set something you return the current value of `front_controller` that not valid with accessor pattern where `set` couldn't return the current value. 
Otherwise you're not be able to do `$this->setControllerType()->setRequestUri()->getData()`.

If not you don't need to return the setting value, you can directly use `$this->front_controller` :)